datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,348 @@
+    for past_flag in h.multi_split(data.pop("Flags (former)"), [" /"]):
+        vessel.add("pastFlags", past_flag)
+    vessel.add("topics", "poi")
+
+    sanction = h.make_sanction(context, vessel)
+    sanction.add("country", data.pop("Sanctions", None))
+    sanction.add("sourceUrl", link)
+
+    context.emit(vessel, target=True)
+    context.emit(sanction)
+
+    linked_entity_name = data.pop(
+        "The person in connection with whom sanctions have been applied"
+    )
+    if linked_entity_name != "":
+        linked_entity = context.make("LegalEntity")
+        linked_entity.id = context.make_id(linked_entity_name)
+        for name in h.multi_split(linked_entity_name, [" | "]):
+            linked_entity.add("name", name)
+        linked_entity.add("topics", "poi")
+        context.emit(linked_entity, target=True)
+
+    crawl_ship_relation(
+        context,
+        vessel,
+        data,
+        "Commercial ship manager (IMO / Country / Date)",
+        "Commercial ship manager",
+        "Representation",
+    )
+    crawl_ship_relation(
+        context,
+        vessel,
+        data,
+        "Ship Safety Management Manager (IMO / Country / Date)",
+        "Ship Safety Management Manager",
+        "Representation",
+    )
+    crawl_ship_relation(
+        context,
+        vessel,
+        data,
+        "Shipowner (IMO / Country / Date)",
+        "Shipowner",
+        "Ownership",
+    )
+    context.audit_data(
+        data,
+        ignore=[
+            "Cases of AIS shutdown",
+            "Calling at russian ports",
+            "Visited ports",
+            "Builder (country)",
+        ],
+    )
+
+
+def crawl_ship_relation(context, vessel, data, data_key, rel_role, rel_schema):
+    # Extract the relation information from data using the specified key
+    relation_info = data.pop(data_key, None)
+    if relation_info is None:
+        return
+    # Split the relation info into expected parts
+    relation_parts = relation_info.split(" / ")
+    if len(relation_parts) == 3:
+        entity_name_imo, entity_country, entity_date = relation_parts
+        if len(h.multi_split(entity_name_imo, [" (", "c/o"])) != 2:
+            overrides = lookup_override(context, entity_name_imo)
+            entity_name = overrides.get("name")
+            entity_imo = overrides.get("registrationCode")
+        else:
+            entity_name, entity_imo = entity_name_imo.split(" (")
+
+        # Create and emit the Legal Entity
+        entity = context.make("LegalEntity")
+        entity.id = context.make_id(entity_name, entity_country)
+        entity.add("name", entity_name)
+        entity.add("registrationNumber", entity_imo)
+        entity.add("country", entity_country)
+        entity.add("topics", "poi")
+        context.emit(entity, target=True)
+
+        # Create the relation representation
+        relation = context.make(rel_schema)
+        relation.id = context.make_id(vessel.id, f"{rel_role} by", entity.id)
```

Comment:
Could also just be this

```suggestion
        relation.id = context.make_id(vessel.id, rel_role, entity.id)
```